diff --git a/src/slope/slope.cpp b/src/slope/slope.cpp
index 5acb6e4..b64faa6 100644
--- a/src/slope/slope.cpp
+++ b/src/slope/slope.cpp
@@ -15,6 +15,7 @@
 #include "utils.h"
 #include <Eigen/Core>
 #include <Eigen/SparseCore>
+#include <RcppEigen.h>
 #include <memory>
 #include <numeric>
 #include <set>
@@ -181,6 +182,10 @@ Slope::path(T& x,
       // TODO: Return a warning code if the solver does not converge
       assert(it < this->max_it - 1 && "Exceeded maximum number of iterations");

+      if (it % 100) {
+        Rcpp::checkUserInterrupt();
+      }
+
       // Compute primal, dual, and gap
       residual = loss->residual(eta, y);
       updateGradient(gradient,
